server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -3343,28 +3346,87 @@ public SecurityOperation getSecurityOperation() {
     }
   }
 
-  // remove any meta entries after a rolled log is no longer referenced
-  Set<DfsLogger> closedLogs = new HashSet<>();
+  // This is a set of WALs that are closed but may still be referenced byt tablets. A LinkedHashSet
+  // is used because its very import to know the order in which WALs were closed when deciding if a
+  // WAL is eligible for removal.
+  LinkedHashSet<DfsLogger> closedLogs = new LinkedHashSet<>();
```

Comment:
Might want to emphasize that this insertion ordering assumes we only have one log open (and thus one closed) at a time. This isn't future proof if we add multiple open logs, but as long as we declare that assumption, I think it's fine.